ICAFIX/scripts/functionhighpassandvariancenormalize.m

```diff
@@ -132,11 +136,18 @@ function functionhighpassandvariancenormalize(TR,hp,fmri,WBC,varargin)
         save_avw(reshape(confounds',size(confounds,2),1,1,size(confounds,1)),[fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf'],'f',[1 1 1 TR]);
         call_fsl(sprintf(['fslmaths ' fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf -bptf %f -1 ' fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf_hp'],0.5*hp/TR));
 
-        save_avw(reshape(cts,ctsX,ctsY,ctsZ,ctsT),[fmri hpstring '.nii.gz'],'f',[1 1 1 TR]);
+        ctsfull=single(zeros(ctsX*ctsY*ctsZ,ctsT));
+	      ctsfull(ctsmask,:)=cts;
+
+        save_avw(reshape(ctsfull,ctsX,ctsY,ctsZ,ctsT),[fmri hpstring '.nii.gz'],'f',[1 1 1 TR]); clear ctsfull;
         call_fsl(['fslmaths ' fmri hpstring '.nii.gz -bptf ' num2str(0.5*hp/TR) ' -1 ' fmri hpstring '.nii.gz']);
```

Comment:
This particular bit of code seems convoluted and very inefficient.  Essentially, we are restoring `cts` from the earlier masking, then writing out an `hp` .nii.gz that hasn't been filtered yet, so that we can call `fslmaths -bptf` to actually filter that file and overwrite itself.  This could all be replaced with a very simple:
`call_fsl(['fslmaths ' fmri '.nii.gz -bptf ' num2str(0.5*hp/TR) ' -1 ' fmri hpstring '.nii.gz']);`

Reply:
A change related to this would be to eliminate the following lines from up front in the code 
```
%% Load volume time series
if dovol > 0
    cts=single(read_avw([fmri '.nii.gz']));
    ctsX=size(cts,1); ctsY=size(cts,2); ctsZ=size(cts,3); ctsT=size(cts,4); 
    cts=reshape(cts,ctsX*ctsY*ctsZ,ctsT);
end
```
and just load and compute/apply the mask immediately prior to the `cts=detrendpoly(cts',hp)';` line, and the `cts=demean(cts')';` line.

Then, just directly filter the input in the `hp>0`, and follow up with the loading/ mask compute and apply after the filtering (as you've already added).